### PR TITLE
Add JSdocs comments for Creator form methods

### PR DESCRIPTION
**Pull request by dpmittal**
### Problem

Documentation is a very important part in Open source project which is not present in the document and it makes it difficult for new developers to collaborate as well as existing developers to remember what they wrote and why they wrote it. 

### Solution

Adding JSdocs comments to methods and their parameters and return values make the code understandable.

### Areas of Impact

JSdocs comments will not affect any part of the codebase

### CODE REVIEW

File: src/client/components/forms/creator.js [hunk 1]
```diff
@@ -55,14 +70,25 @@ class CreatorForm extends React.Component {
...
+	/**
+	* Handler which directs the viewer to the next tab
+	*/
 	handleNextClick() {
 		this.handleTabSelect(this.state.tab + 1);
 	}
 
+	/**
+	* Handler which collects the creator information edited by the author
+	*	and sends it to the database to be revised.
```
Comment (LordSputnik) [inline src/client/components/forms/creator.js]: "sends it to the server to be processed" would be better here.

File: src/client/components/forms/creator.js [hunk 2]
```diff
@@ -29,7 +30,16 @@ const CreatorData = require('./parts/creator-data');
...
 
+/**
+* This is an extended class of the React Component to
+* create and edit a creator
+*/
 class CreatorForm extends React.Component {
+	/**
+	* @constructor
+	* @param {object} props - Object of properties passed in the constructor.
+	* Binds the class methods to their respective data
```
Comment (LordSputnik) [inline src/client/components/forms/creator.js]: Please put this description above the `@constructor` and `@param` lines

File: src/client/components/forms/creator.js [hunk 3]
```diff
@@ -111,6 +137,12 @@ class CreatorForm extends React.Component {
 			});
 	}
 
+	/**
+	* Renders the document and allows the author to submit the data
+	* Includes a loading spinner if there is delay in data display due to
+	* connectivity.
+	* @return {HTML} a HTML document which displays a creatoreditor
```
Comment (LordSputnik) [inline src/client/components/forms/creator.js]: Actually returns a React Element - let's use the `ReactElement` type to indicate this from now on.

**Comment by dpmittal** [comment]
@LordSputnik I have made the changes :D

**Pull request merged**